app/src/lib/stores/repositories-store.ts

```diff
@@ -399,6 +404,26 @@ export class RepositoriesStore extends BaseStore {
           gitHubRepositoryID: updatedGitHubRepo.dbID,
         })
 
+        const repoId = updatedGitHubRepo.dbID!
+
+        await this.db.protectedBranches
```

Comment:
Don't we have to clear the cache at this point? Say that the last time we fetched `branch-a` was protected but now it no longer is we wouldn't update its entry in the cache since it won't be returned in the api response.

Reply:
That's a case I didn't think about initially - good call. The problem I have currently is the cache key is built off the repository id (known) and the branch name (unknown). I guess I could enumerate the keys and filter on everything starting with a certain pattern - will investigate.

Reply:
Addressed in e36df6dd69209202e433f28fb5aa9224cb0c03fb